api/v1alpha1/virtualmachine_types.go

```diff
@@ -320,8 +320,14 @@ type VirtualMachineSpec struct {
 	// instance.  See VirtualMachineClass for more description.
 	ClassName string `json:"className"`
 
-	// PowerState describes the desired power state of a VirtualMachine.  Valid power states are "poweredOff" and "poweredOn".
-	PowerState VirtualMachinePowerState `json:"powerState"`
+	// PowerState describes the desired power state of a VirtualMachine.
+	//
+	// Please note this field may be omitted when creating a new VM and will
+	// default to "poweredOn." However, once the field is set to a non-empty
+	// value, it may no longer be set to an empty value.
+	//
+	// +optional
+	PowerState VirtualMachinePowerState `json:"powerState,omitempty"`
```

Comment:
From a backward compatibility POV, changing a mandatory field to an optional one should be allowed (since we are making the API more relaxed).  But can this be considered a behavior change -- since this technically breaks any automation that relies on the webhook returning error for a spec that did not contain `powerState`?

Reply:
Fair point. But, realistically, the response to such an error would be to correct it, right? The only impact is if it was corrected to poweredOff in someone's logic. I think the likelyhood for impact is minimal to none and worth the value it adds to the overall UX. Thoughts?

Reply:
Yeah, fair enough.  The value add here is definitely substantial compared to the off chance that someone has built an automation around the behavior of our webhooks.